version: '3.8'
services:
  db:
    image: postgis/postgis:13-3.1
    platform: linux/amd64
    ports:
      - "5432:5432"
    volumes:
      - $PWD/pgdata:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "postgres"]
      interval: 5s
      retries: 5

  django:
    container_name: django
    build:
      context: ./backend
    command: python manage.py runserver 0.0.0.0:8000
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ./backend:/code
    restart: always
  
  # client:
  #   container_name: client
  #   build: 
  #     context: ./Client
  #   command: npm start
  #   ports:
  #     - "3000:3000"
  #   volumes:
  #     - ./Client:/app
  #   restart: always
